fdbserver/FDBExecHelper.actor.cpp

```diff
@@ -400,7 +400,17 @@ ACTOR Future<int> execHelper(ExecCmdValueString* execArg, UID snapUID, std::stri
 	state int err = 0;
 	state Future<int> cmdErr;
 	state double maxWaitTime = SERVER_KNOBS->SNAP_CREATE_MAX_TIMEOUT;
+	// split roles and folders if one process running as multiple roles
+	state std::vector<std::string> roles;
+	boost::algorithm::split(roles, role, boost::is_any_of(","));
+	state std::vector<std::string> folders;
+	boost::algorithm::split(folders, folder, boost::is_any_of(","));
+	ASSERT(roles.size() == folders.size());
+	ASSERT(roles.size() >= 1 && roles.size() <= 3);
 	if (!g_network->isSimulated()) {
+		// if not simulated, only use the first one
+		role = roles[0];
+		folder = folders[0];
```

Comment:
Does this mean that outside simulation it is not possible to snapshot (e.g.) the coordinator and storage roles together?

Reply:
I will change back to the case where we sent requests to each role